crates/anvil/tests/it/anvil.rs

```diff
@@ -118,3 +121,20 @@ async fn test_cancun_fields() {
     assert!(block.header.blob_gas_used.is_some());
     assert!(block.header.excess_blob_gas.is_some());
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_cache_path() {
+    run_with_tempdir("custom-anvil-cache", |tmp_dir| async move {
+        let cache_path = tmp_dir.join("cache");
+        let (_api, _handle) = spawn(
+            NodeConfig::test()
+                .with_cache_path(Some(cache_path.clone()))
+                .with_max_persisted_states(Some(10_usize))
+                .with_blocktime(Some(Duration::from_millis(1))),
+        )
+        .await;
```

Comment:
is there a way to assert file was actually written in custom temp dir? I tried to manually test it but probably I am missing something as I wasn't able to get the json file dumped

Reply:
Yeah so testing this is a bit hard. Wrote a better test here. [363010f](https://github.com/foundry-rs/foundry/pull/9343/commits/363010fe48c4e93085e1f5537626404d71b1e5e5)

Reply:
thank you! yeah, managed to check file is generated with this test. I think that's good to have, drawback is that will slow down the CI (and potential flakiness), @zerosnacks wdyt?

If we are to keep it let's solve warnings here
https://github.com/foundry-rs/foundry/actions/runs/11933626813/job/33261072474#step:12:66

and maybe add one more assert to make sure the cache is actually populated
```rust
assert!(cache_path.read_dir().unwrap().next().is_some());
```